### PR TITLE
Allow registry node to connect peers from registry

### LINKED ISSUE #2482: Registry should be able to start without a network peer
When we start a `FullNode`, it expects to either know about a registry or have statically defined network peers.
However, if the `FullNode` *is* the registry, we have a chicken-and-egg problem. Hence, a FullNode should be able to start without any known peer provided it's a registry.

Part of #2256

### PR DESCRIPTION
**Pull request by mkykadir**
Fixes #2482

### CODE REVIEW

File: source/agora/node/FullNode.d [hunk 1]
```diff
@@ -396,7 +396,7 @@ public class FullNode : API
                 .each!(handler => handler.addresses
                     .each!((string address) {
                         auto url = Address(address); // TODO normalize
-                        this.block_header_handlers[url] = this.network.getBlockHeaderUpdatedHandler(url);
```
Comment (Geod24) [inline source/agora/node/FullNode.d]: Whitespace fixes in their own commit.

File: source/agora/network/Manager.d [hunk 2]
```diff
@@ -760,6 +761,17 @@ public class NetworkManager
             return true;
         }
 
+        try
+        {
+            auto registry = (cast(FullNode) this.owner_node).getRegistry();
+            if (registry !is null)
+                foreach(addr; registry.getValidatorsAddresses())
+                    this.addAddress(addr);
```
Comment (Geod24) [inline source/agora/network/Manager.d]: It's a bit bothering that we would:
- `cast` an interface to a `class` type (that's a failure of encapsulation);
- Expose an internal member of a class via `public` (I already wasn't happy with `package`);

File: source/agora/node/FullNode.d (hunk 1 above)
Comment (mkykadir) [reply source/agora/node/FullNode.d]: Editor did it automatically on save, I will revert

File: source/agora/network/Manager.d (hunk 2 above)
Comment (omerfirmak) [reply source/agora/network/Manager.d]: You could easily pass these to `discover()`, as parameters (which is already called by the `owner_node`). No need to cast or expose internals.

File: source/agora/node/Registry.d [hunk 3]
```diff
@@ -756,6 +776,16 @@ private struct ZoneData
         return typed_payload;
     }
 
+    public Address[] getAddresses ()
+    {
+        auto results = this.db.execute(this.query_addresses_get);
+
+        if (results.empty)
+            return new Address[0];
```
Comment (Geod24) [inline source/agora/node/Registry.d]: `return null` ?

**Review by Geod24 — APPROVED** [review]
LGTM, just the return, and then feel free to self merge

**Pull request merged**